carbon-design-system/ibm-security
# fix: modular styles for proxied components

Updates stylesheets for components styles that are "proxied" from `carbon-components`. In most cases this means adding the following local mixins:

```scss

// Security variables, notably, unlike Carbon it turns off reset
@import '../../globals/variables/index';

// The proxied component
@import 'carbon-components/scss/components/SOME-COMPONENT/SOME-COMPONENT';

// Security mixin to add a namespace
@import '../../globals/namespace/index';
```

## Affected issues

- Relates to #44 

## Proposed changes

Makes the above updates to the following components
- Checkbox
- CodeSnippet
- ContentSwitcher
- Dropdown (isolated style issues, suspected with Carbon / upstream)
- Form
- InlineLoading
- Link
- Modal
- MuliSelect
- NumberInput
- OverflowMenu
- Pagination
- RadioButton
- Select
- Slider
- StructuredList
- Tabs
- TextArea
- TextInput
- TimePicker (isolated style issues, suspected with Carbon / upstream)

## Testing instructions

- Feel free to pull this repo down and add `import './_index.scss';` to each respective story in this PR...
- All components behave as expected except for TimePicker and Dropdown. More investigation needed here but I am thinking this is an issue with `carbon-components` and not on our end
